pyanaconda/modules/localization/live_system.py

```diff
@@ -0,0 +1,99 @@
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from abc import ABC, abstractmethod
+from pyanaconda.core.util import execWithCaptureAsLiveUser
+from pyanaconda.core.configuration.anaconda import conf
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+def get_live_keyboard_instance():
+    """Return instance of a class based on the current running live system.
+
+    :return: instance of a LiveSystemKeyboardBase based class for the current live environment
+    :rtype: instance of class inherited from LiveSystemKeyboardBase class
+    """
+    if conf.system.provides_liveuser:
+        return GnomeShellKeyboard()
+
+    # TODO: Add support for other Live systems
+    return None
+
+
+class LiveSystemKeyboardBase(ABC):
+
+    @abstractmethod
+    def read_keyboard_layouts(self):
+        """Read keyboard configuration from the current system.
+
+        The configuration have to be returned in format which is understandable by us for the
+        installation. That means localed will understand it.
+
+        :return: a list of "layout (variant)" or "layout" layout specifications
+        :rtype: list(str)
+        """
+        pass
+
+    @staticmethod
+    def _run_as_liveuser(argv):
+        """Run the command in a system as liveuser user.
+
+        :param list argv: list of arguments for the command.
+        :return: output of the command
+        :rtype: str
+        """
+        return execWithCaptureAsLiveUser(argv[0], argv[1:])
+
+
+class GnomeShellKeyboard(LiveSystemKeyboardBase):
+
+    def read_keyboard_layouts(self):
+        """Read keyboard configuration from the current system.
+
+        The configuration have to be returned in format which is understandable by us for the
+        installation. That means localed will understand it.
+
+        :return: a list of "layout (variant)" or "layout" layout specifications
+        :rtype: list(str)
+        """
+        command_args = ["gsettings", "get", "org.gnome.desktop.input-sources", "sources"]
+        sources = self._run_as_liveuser(command_args)
+
+        # convert output "[('xkb', 'us'), ('xkb', 'cz+qwerty')]\n"
+        try:
+            sources = eval(sources.rstrip())
+        except SyntaxError:
+            log.error("Gnome Shell keyboard configuration can't be obtained from source %s!",
+                      sources)
+            return []
+
+        result = []
+
+        for t in sources:
+            # keep only 'xkb' type and ignore 'ibus' variants which can't be used in localed
+            if t[0] == "xkb":
+                layout = t[1]
+                # change layout variant from 'cz+qwerty' to 'cz (qwerty)'
+                if '+' in layout:
```

Comment:
It might be useful to mention nearby what the expected string looks like.

Reply:
It is as the first comment in the function.

Reply:
Improved the comment to add also expected output of the function.